middlewares/selectCourses.sql

```diff
@@ -0,0 +1,23 @@
+-- BLOCK select_courses
+WITH
+    course_permissions_for_user AS (
+        SELECT
+            *
+        FROM
+            course_permissions AS cp
+        WHERE
+            cp.user_id = $user_id
+    )
+SELECT
+    c.*
+FROM
+    pl_courses AS c
+    LEFT JOIN course_permissions_for_user AS cp ON (cp.course_id = c.id)
```

Comment:
```suggestion
    LEFT JOIN course_permissions AS cp ON (cp.course_id = c.id AND cp.user_id = $user_id)
```

With this change you can delete the `WITH` block.

Reply:
Done/outdated